src/test/java/io/vertx/ext/dropwizard/MetricsTest.java

```diff
@@ -354,14 +354,15 @@ public void testHttpMetricsOnClose() throws Exception {
     client.close();
     server.close(ar -> {
       assertTrue(ar.succeeded());
-      vertx.runOnContext(v -> testComplete());
+      // This allows http server metrics to be completely removed from the registry
+      vertx.setTimer(100, v -> testComplete());
```

Comment:
this should rather loop until `metricsService.getMetricsSnapshot(server).getMap().isEmpty()` and a timer < 10000 millis:

```
AtomicInteger count = new AtomicInteger(100);
vertx.setPeriodic(100, id -> {
  if (count.decrementAndGet() == 0) {
     vertx.cancelPeriodic(id);
     fail();
  } else if (metricsService.getMetricsSnapshot(server).getMap().isEmpty()) {
     vertx.cancelPeriodic(id);
     testComplete();
  }
});
```